utils/update_build_version.py

```diff
@@ -50,6 +54,23 @@ def command_output(cmd, dir):
     return stdout
 
 
+def deduceSoftwareVersion(dir):
+    """Returns a software version number parsed from the CHANGES file
+    in the given dir.
+
+    The CHANGES file describes most recent versions first.
+    """
+
+    pattern = re.compile('(v\d+\.\d+(wip)?) ')
```

Comment:
If we expect the entire WIP line to be of the form

v2016.0wip 2016-04-21

Then it might be nice to change the pattern to make sure that the entire line is correct.

$(v\d+\.\d+(wip)? \d\d\d\d-\d\d-\d\d^
Also if we expect that to always be the first line after "Revision history for SPIRV-Tools" then we could error out if the second non-empty line does not have that format. It feels like that would catch mistakes in the file more easily.